oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -799,15 +782,19 @@ impl Runtime {
     /// Waits on a slice of `ChannelHalf`s, blocking until one of the following conditions:
     /// - If the [`Runtime`] is terminating this will return immediately with an `ErrTerminated`
     ///   status for each channel.
-    /// - If all readers are in an erroneous status, e.g. when all channels are orphaned, this will
-    ///   immediately return the channels statuses.
+    /// - If any of the readers is in an erroneous status, e.g. when a channel is orphaned, this
+    ///   will immediately return with all the channels statuses set in the returned vector.
```

Comment:
(pre-existing) Maybe reword the entire comment, since it kind of contradicts itself saying that the method blocks until one of these conditions. Make it clear under what circumstances it returns immediately and when it blocks, and when blocked, what condition(s) unblock it, and what is the return value for each case.